### PR TITLE
changelog: update changelog for v0.34.1

### PR DESCRIPTION
**Pull request by tessr**

### CODE REVIEW

**Review by erikgrinaker — COMMENTED** [review]
This changelog entry should be moved as well:

>- [libs/protoio] Return number of bytes read in `Reader.ReadMsg()` (@erikgrinaker)

Note that this is a breaking change to a supported Go API, but we say in the README that that's fine for patch releases.

**Comment by tessr** [comment]
What's the PR # associated with that change?

**Comment by erikgrinaker** [comment]
Gah, sorry, forgot to add it. #5868, same as the P2P fix.

**Comment by tessr** [comment]
Ahhh, got it. OK. Hmm, do you think this is likely to break anyone?

**Comment by tessr** [comment]
I guess we also are changing the CLI; maybe we can just write this off as another pre-1.0 shenanigan.

**Comment by erikgrinaker** [comment]
Well, the current policy is that we're allowed to break Go APIs in patch releases, as long as it's documented in the changelog. Just mentioning it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,9 @@ Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermi
 - CLI/RPC/Config
   - [cli] [\#5786](https://github.com/tendermint/tendermint/issues/5786) deprecate snake_case commands for hyphen-case (@cmwaters)
 
+- Go API
+  - [libs/protoio] [\#5868](https://github.com/tendermint/tendermint/issues/5868) Return number of bytes read in `Reader.ReadMsg()` (@erikgrinaker
```
Comment (erikgrinaker) [inline CHANGELOG.md:23]:
```suggestion
  - [libs/protoio] [\#5868](https://github.com/tendermint/tendermint/issues/5868) Return number of bytes read in `Reader.ReadMsg()` (@erikgrinaker)
```

**Review by erikgrinaker — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (tessr) [reply CHANGELOG.md:23]: 🤦🏼  THANK YOU.

**Comment by cmwaters** [comment]
Minor note: This CLI commit isn't exactly breaking per se. It keeps all the existing commands just adds a deprecation warning when they are run and allows users to switch to the hypen-case format. Maybe I should have put this under a different section

**Pull request merged**